examples/cache-handler-redis/cache-handler.js

```diff
@@ -1,44 +1,77 @@
-  url: process.env.REDIS_URL ?? "redis://localhost:6379",
-});
-
-client.on("error", (error) => {
-  console.error("Redis error:", error.message);
-});
-
+/* from https://caching-tools.github.io/next-shared-cache/redis */
 CacheHandler.onCreation(async () => {
-  // read more about TTL limitations https://caching-tools.github.io/next-shared-cache/configuration/ttl
-  function useTtl(maxAge) {
-    const evictionAge = maxAge * 1.5;
+  let client;
+  // use redis client during build could cause issue https://github.com/caching-tools/next-shared-cache/issues/284#issuecomment-1919145094
+  if (PHASE_PRODUCTION_BUILD !== process.env.NEXT_PHASE) {
+    try {
+      // Create a Redis client.
+      client = createClient({
+        url: process.env.REDIS_URL ?? "redis://localhost:6379",
+      });
 
-    return evictionAge;
+      // Redis won't work without error handling.
+      client.on("error", (e) => {
+        throw e;
+      });
+    } catch (error) {
+      console.warn("Failed to create Redis client:", error);
+    }
   }
 
-  let redisCache;
+  if (client) {
+    try {
+      console.info("Connecting Redis client...");
 
-  if (PHASE_PRODUCTION_BUILD !== process.env.NEXT_PHASE) {
-    await client.connect();
+      // Wait for the client to connect.
+      // Caveat: This will block the server from starting until the client is connected.
+      // And there is no timeout. Make your own timeout if needed.
+      await client.connect();
+      console.info("Redis client connected.");
+    } catch (error) {
+      console.warn("Failed to connect Redis client:", error);
+
+      console.warn("Disconnecting the Redis client...");
+      // Try to disconnect the client to stop it from reconnecting.
+      client
+        .disconnect()
+        .then(() => {
+          console.info("Redis client disconnected.");
+        })
+        .catch(() => {
+          console.warn(
+            "Failed to quit the Redis client after failing to connect.",
+          );
+        });
+    }
+  }
 
-    redisCache = await createRedisCache({
+  /** @type {import("@neshca/cache-handler").Handler | null} */
+  let handler;
+
+  if (client?.isReady) {
+    // Create the `redis-stack` Handler if the client is available and connected.
+    handler = await createRedisHandler({
       client,
-      useTtl,
+      keyPrefix: "prefix:",
+      timeoutMs: 1000,
     });
+  } else {
+    // Fallback to LRU handler if Redis client is not available.
+    // The application will still work, but the cache will be in memory only and not shared.
+    handler = createLruHandler();
+    console.warn(
+      "Falling back to LRU handler because Redis client is not available.",
+    );
   }
 
-  const localCache = createLruCache({
-    useTtl,
-  });
-
   return {
-    cache: [redisCache, localCache],
-    // read more about useFileSystem limitations https://caching-tools.github.io/next-shared-cache/configuration/use-file-system
-    useFileSystem: false,
+    handlers: [handler],
```

Comment:
i think its helpful to show setting 2 handlers
```
handlers: [redisCache, localCache],
```

Reply:
@papertiger12  Sure. I've just updated the handlers's array to use both handlers.  Thanks for the feedback